crates/utils/src/settings.rs

```diff
@@ -69,11 +119,34 @@ pub struct DatabaseConfig {
   pub pool_size: u32,
 }
 
+impl Default for DatabaseConfig {
+  fn default() -> Self {
+    Self {
+      user: "lemmy".into(),
+      password: "password".into(),
+      host: "localhost".into(),
+      port: 5432,
+      database: "lemmy".into(),
+      pool_size: 5,
+    }
+  }
+}
+
 #[derive(Debug, Deserialize, Clone)]
 pub struct FederationConfig {
   pub enabled: bool,
-  pub allowed_instances: String,
-  pub blocked_instances: String,
+  pub allowed_instances: Option<String>,
+  pub blocked_instances: Option<String>,
```

Comment:
These should be `Vec<String>`, not sure why I implemented it like this.

Reply:
With this hjson-deser, these must be options, otherwise you'd have always have to provide both fields.

Reply:
Then `Option<Vec<String>>`

Reply:
Ohhh.... assuming `deser_hjson` can handle vectors (I think it can fine) that would be a breaking change, because then the config files would have to look like:

`allowed_instances: [instance1, instance2,...]`

Right now they are 

`allowed_instances: instance1,instance2`

Reply:
We can just publish this with Lemmy v0.10.0 then the breaking change is okay. Of course it would be nice if we could make an automatic migration somehow, but either way we should do that.

Unless we plan to move these vars into the database, which would be necessary for #862.